modular_skyrat/modules/goofsec/jobs/sol_fed.dm

```diff
@@ -224,6 +317,150 @@ GLOBAL_LIST_INIT(emergency_responders, list())
+		activated = TRUE
+		icon_state = "reporter_on"
+		GLOB.votes_for_treason++
+		to_chat(user, span_warning("You have reported the station as participating in Treason. \
+		Current Votes: [GLOB.votes_for_treason]/[GLOB.amt_SWAT_responders] votes."))
+		if(GLOB.votes_for_treason >= GLOB.amt_SWAT_responders)
+			to_chat(user, span_warning("The station has been declared as engaging in Treason!"))
+			GLOB.treason_declared = TRUE
+
+			priority_announce("Crewmembers of [station_name()].\n\
+			You have refused to comply with first responders and SWAT officers, and have assaulted them, and they are unable to \
+			carry out the wills of the Sol Federation, despite residing within Sol Federation borders.\n\
+			As such, we are charging those responsible with Treason. The penalty of which is death, or no less than twenty-five years in Superjail.\n\
+			Treason is a serious crime. Our military forces are en route to your station. They will be assuming direct control of the station, and \
+			will be evacuating civilians from the scene.\n\
+			Comply, or be shot.\n\
+			Non-offending citizens, prepare for evacuation. Comply with all orders given to you by Sol Federation military personnel.\n\
+			To all those who are engaging in treason, lay down your weapons and surrender. Lethal force has been authorized if you do not comply.\n\
+			Have a secure day.", \
+			"Sol Federation Military", 'sound/effects/families_police.ogg', has_important_message=TRUE)
+			var/list/candidates = poll_ghost_candidates("The station has decided to engage in treason. Do you wish to join the Sol Federation Military?", "deathsquad")
+
+			if(candidates.len)
+				//Pick the (un)lucky players
+				var/numagents = min(12,candidates.len)
+
+				var/list/spawnpoints = GLOB.emergencyresponseteamspawn
+				var/index = 0
+				while(numagents && candidates.len)
+					var/spawnloc = spawnpoints[index+1]
+					//loop through spawnpoints one at a time
+					index = (index + 1) % spawnpoints.len
+					var/mob/dead/observer/chosen_candidate = pick(candidates)
+					candidates -= chosen_candidate
+					if(!chosen_candidate.key)
+						continue
+
+					//Spawn the body
+					var/mob/living/carbon/human/cop = new(spawnloc)
+					chosen_candidate.client.prefs.safe_transfer_prefs_to(cop, is_antag = TRUE)
+					cop.key = chosen_candidate.key
+
+					//Give antag datum
+					var/datum/antagonist/ert/request_911/ert_antag = new /datum/antagonist/ert/request_911/treason_destroyer
+
+					cop.mind.add_antag_datum(ert_antag)
+					cop.mind.set_assigned_role(SSjob.GetJobType(ert_antag.ert_job_path))
+					SSjob.SendToLateJoin(cop)
+
+					var/obj/item/gangster_cellphone/phone = new() // biggest gang in the city
+					phone.gang_id = "911"
+					phone.name = "911 branded cell phone"
+					var/phone_equipped = phone.equip_to_best_slot(cop)
+					if(!phone_equipped)
+						to_chat(cop, "Your [phone.name] has been placed at your feet.")
+						phone.forceMove(get_turf(cop))
```

Comment:
These just all feel like they could be simplified using inheritance and variables for the content of those messages, etc.